webdriver/tests/support/helpers.py

```diff
@@ -60,24 +60,37 @@ def _restore_window_state(session):
             session.window.size = defaults.WINDOW_SIZE
 
     @ignore_exceptions
-    def _restore_windows(session):
+    async def _restore_windows(session):
         """Close superfluous windows opened by the test.
 
         It will not end the session implicitly by closing the last window.
+        If available, uses WebDriver BiDi session. Otherwise, uses the WebDriver Classic.
         """
-        current_window = session.window_handle
 
-        for window in _windows(session, exclude=[current_window]):
-            session.window_handle = window
-            if len(session.handles) > 1:
-                session.window.close()
+        if session.bidi_session:
+            response = await (
+                await session.bidi_session.send_command("browsingContext.getTree", {
+                    "maxDepth": 1
+                }))
+            for context in response["contexts"][1:]:
```

Comment:
question: So here we are closing all windows except the first one, which assumes that `session.window_handle` is the first window? Is it really a safe assumption?

Reply:
Fair point. Updated the logic, so that if the `session.window_handle` is still open, it is kept open. Otherwise `session.window_handle` is set to the first open browsing context.